src/r_data.c

```diff
@@ -848,6 +848,30 @@ void R_InitSpriteLumps(void)
 // killough 4/4/98: Add support for C_START/C_END markers
 //
 
+static byte invul_orig[256];
+
+void R_InvulMode(void)
+{
+  if (colormaps == NULL)
+    return;
+
+  switch (STRICTMODE(invul_mode))
+  {
+    case INVUL_VANILLA:
+      default_comp[comp_skymap] = 1;
+      memcpy(&colormaps[0][256*32], invul_orig, 256);
+      break;
+    case INVUL_MBF:
+      default_comp[comp_skymap] = 0;
+      memcpy(&colormaps[0][256*32], invul_orig, 256);
+      break;
+    case INVUL_GRAY:
+      default_comp[comp_skymap] = 1;
```

Comment:
How do `comp_skymap` and `invul_mode` interact? Are both necessary or could it be one setting?

Reply:
`invul_mode` changes the `comp_skymap` setting. We can make them independent of each other.